src/Services/OData/ODataFormatter.cs

```diff
@@ -510,6 +510,7 @@ private object ProcessOperationResponse(object response, PortalContext portalCon
                 Top = req.Top > 0 ? req.Top : qdef.Top,
                 Skip = req.Skip > 0 ? req.Skip : qdef.Skip,
                 Sort = req.Sort != null && req.Sort.Count() > 0 ? req.Sort : qdef.Sort,
+                AllPages = req.InlineCount == InlineCount.AllPages,

```

Comment:
There is a HasInlineCount property of ODataRequest, maybe we should check that and if it is not set, than use the value from qdef, like in case of other properties in this method.